### PR TITLE
Updated zenoh-c to latest commit

### PR DESCRIPTION
**Pull request by imstevenpmwork**
Hi @Yadunund!
We're submitting this PR to ensure we use the latest zenoh-c commit before proceeding with our planned testing

### CODE REVIEW

**Review by Yadunund — CHANGES_REQUESTED** [review]
Could you also update the comment above the with PR # associated with this commit hash?

File: zenoh_c_vendor/CMakeLists.txt [hunk 1]
```diff
@@ -24,11 +24,10 @@ find_package(ament_cmake_vendor_package REQUIRED)
 # when expanded.
 set(ZENOHC_CARGO_FLAGS "--no-default-features$<SEMICOLON>--features=zenoh/transport_tcp zenoh/shared-memory")
 
-# Set VCS_VERSION to include changes from https://github.com/eclipse-zenoh/zenoh/pull/875
-# which was synced to zenoh-c in https://github.com/eclipse-zenoh/zenoh-c/pull/307.
+# Set VCS_VERSION to include latest changes from zenoh-c https://github.com/eclipse-zenoh/zenoh-c/pull/340
```
Comment (Yadunund) [inline zenoh_c_vendor/CMakeLists.txt]:
```suggestion
# Set VCS_VERSION to include latest changes from zenoh-c https://github.com/eclipse-zenoh/zenoh-c/pull/340.
```

**Review by Yadunund — COMMENTED** [review]
Sorry minor nitpick.

**Comment by clalancette** [comment]
I think I'm a little confused by this PR.  As far as I understood, if we bump to the latest `zenoh-c`, then some of our tests stop working.  We are still debugging that.  Am I misunderstanding, or did we solve that particular issue?

**Comment by imstevenpmwork** [comment]
Hello @clalancette! This pull request targets the branch to evaluate rmw with Tokio zenoh-c. Although the current hash already corresponds to Tokio zenoh-c, it's a few weeks outdated. Hence, we believe it's prudent to update this branch to the latest Tokio zenoh-c version and rerun the tests to assess any potential improvements or regressions.

**Review by Yadunund — APPROVED** [review]
LGTM.

**Pull request merged**